bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_DataEngine.cpp

```diff
@@ -100,6 +101,30 @@ namespace sofapython3
         throw py::type_error(this->getName() + " has no update() method.");
     }
 
+
+    py::list property_inputs(DataEngine* self)
+    {
+        py::list list;
+        auto fields = self->getDataFields();
+        auto it = std::remove_if(fields.begin(), fields.end(), [&](const auto & data){ return data->getGroup() != "Inputs"; });
+        fields.erase(it, fields.end());
+        for(auto i : fields)
+            list.append(PythonFactory::toPython(i));
+        return list;
+    }
+
+    py::list property_outputs(DataEngine* self)
+    {
+        py::list list;
+        auto fields = self->getDataFields();
+        auto it = std::remove_if(fields.begin(), fields.end(), [&](const auto & data){ return data->getGroup() != "Inputs"; });
```

Comment:
Wow, did not know `std::remove_if` was a thing, that's handy. Why the erase after it tho? To resize the iterator?

Reply:
It's actually very wierd :) remove_if removes the elements that fulfill the condition, but do not resize the container instead if "recompacts" the values and leaves the last entries of the container empty, returning the iterator to the last "non-empty" element.
Thus once removed, it is still up to you to call erase to remove blank cells and resize the container

Reply:
That's a funny design haha, thanks !